sklearn/decomposition/_online_lda.pyx

```diff
@@ -0,0 +1,73 @@
+cimport cython
+cimport numpy as np
+import numpy as np
+
+np.import_array()
+
+cdef np.float64_t NEGINF = -np.inf
+
+from libc.math cimport fabs
+from scipy.special import psi
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def mean_change(np.ndarray[ndim=1, dtype=np.float64_t] arr_1,
+               np.ndarray[ndim=1, dtype=np.float64_t] arr_2):
+    
+    cdef np.float64_t total, diff
+    cdef np.npy_intp i, size
+
+    size = arr_1.shape[0]
+    total = 0.0
+    for i in range(size):
+        diff = fabs(arr_1[i] - arr_2[i])
+        total += diff
+
+    return total / size
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def _dirichlet_expectation_1d(np.ndarray[ndim=1, dtype=np.float64_t] arr):
```

Comment:
Again, did you benchmark?
This is ``psi(arr) - psi(arr.sum())``, right?
Can you please add this as documentation?